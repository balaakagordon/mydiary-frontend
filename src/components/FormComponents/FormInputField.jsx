import React from 'react';


function createLabel(props) {
    return (
        <label htmlFor={props.name} className="form-label">{props.title}</label>
    )
}

const FormInputField = (props) => {
    return (
        <div className="form-group">
            {props.title ? createLabel(props) : null}
            <input
                className="form-input"
                id={props.name}
                name={props.name}
                type={props.type}
                defaultValue={props.titleValue ? props.titleValue : null}
                // value={props.value ? props.value : null}
                onChange={props.handleChange}
                placeholder={props.placeholder}
            />
        </div>
    )
}

export default FormInputField;