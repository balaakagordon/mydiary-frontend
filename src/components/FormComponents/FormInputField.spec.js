import React from 'react';
import { mount } from 'enzyme';
import FormInputField from './FormInputField';

describe('The FormInputField component', () => {
    let component;

    beforeAll(() => {
        component = mount(<FormInputField />)
    })

    it('should render without crashing', () => {
        expect(component).toBeDefined();
    })

    it('should match snapshot', () => {
        expect(component).toMatchSnapshot();
    })
})

describe('FormInputField functions', () => {
    let handleChange = jest.fn();

    it('should call handleChange onChange', () => {
        let wrapper = mount(
            <FormInputField
                title={'title'}
                name={'name'}
                value={''}
                handleChange={handleChange}
            />
        )
        wrapper.find('input').simulate('change');
        expect(handleChange).toHaveBeenCalled();
    })
})
