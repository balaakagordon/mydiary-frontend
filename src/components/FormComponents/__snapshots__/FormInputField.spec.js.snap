// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The FormInputField component should match snapshot 1`] = `
<FormInputField>
  <div>
    <div
      className="entry-title"
    >
      <input
        defaultValue={null}
      />
    </div>
  </div>
</FormInputField>
`;
